faucet/src/lib.rs

```diff
@@ -0,0 +1,179 @@
+mod mock;
+#[cfg(test)]
+mod tests;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(PhantomData<T>);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + encointer_reputation_commitments::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		type Currency: Currency<Self::AccountId>;
+		type FaucetMaster: EnsureOrigin<Self::RuntimeOrigin>;
+
+		#[pallet::constant]
+		type PalletId: Get<PalletId>;
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Transfer some balance to another account.
+		#[pallet::call_index(0)]
+		#[pallet::weight(10_000)]
+		pub fn drip(
+			origin: OriginFor<T>,
+			cid: CommunityIdentifier,
+			cindex: CeremonyIndexType,
+		) -> DispatchResultWithPostInfo {
+			let from = ensure_signed(origin)?;
+
+			<encointer_reputation_commitments::Pallet<T>>::do_commit_reputation(
+				&from,
+				cid,
+				cindex,
+				Self::reputation_commitments_purpose_id(),
+				None,
+			)?;
+
+			T::Currency::transfer(&Self::account_id(), &from, Self::drip_amount(), KeepAlive)
+				.map_err(|_| <Error<T>>::FaucetEmpty)?;
+
+			Self::deposit_event(Event::Dripped(from, Self::drip_amount()));
+
+			Ok(().into())
+		}
+
+		#[pallet::call_index(1)]
+		#[pallet::weight(10_000)]
+		pub fn set_drip_amount(
+			origin: OriginFor<T>,
+			drip_amount: BalanceOf<T>,
+		) -> DispatchResultWithPostInfo {
+			T::FaucetMaster::ensure_origin(origin)?;
+			<DripAmount<T>>::put(drip_amount);
+			info!(target: LOG, "set drip amount to {:?}", drip_amount);
+			Self::deposit_event(Event::DripAmountUpdated(drip_amount));
+			Ok(().into())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		/// The account ID of the faucet.
+		///
+		/// This actually does computation. If you need to keep using it, then make sure you cache the
+		/// value and only call this once.
+		pub fn account_id() -> T::AccountId {
+			T::PalletId::get().into_account_truncating()
+		}
+
+		pub fn pot() -> BalanceOf<T> {
+			T::Currency::free_balance(&Self::account_id())
+				// Must never be less than 0 but better be safe.
+				.saturating_sub(T::Currency::minimum_balance())
+		}
+	}
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config> {
+		pub drip_amount: BalanceOf<T>,
+	}
+
+	#[cfg(feature = "std")]
+	impl<T: Config> Default for GenesisConfig<T> {
+		fn default() -> Self {
+			Self { drip_amount: Default::default() }
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {
+			// Create faucet account
+			let account_id = <Pallet<T>>::account_id();
+			let min = T::Currency::minimum_balance();
+			if T::Currency::free_balance(&account_id) < min {
+				let _ = T::Currency::make_free_balance_be(&account_id, min);
+			}
+			<DripAmount<T>>::put(self.drip_amount);
```

Comment:
as DripAmount is in the genesis config, this should not be necessary

Reply:
hm, but this seems like a pattern we use everywhere, eg. [here](https://github.com/encointer/pallets/blob/90b70555507f83fb842a83dbbdee8c8877bf2899/communities/src/lib.rs#L447)

Reply:
ok